scalding-core/src/main/scala/com/twitter/scalding/CumulativeSum.scala

```diff
@@ -39,9 +39,9 @@ object CumulativeSum {
     def cumulativeSum(
       implicit sg: Semigroup[V], ordU: Ordering[U], ordK: Ordering[K]): SortedGrouped[K, (U, V)] = {
       pipe.group
-        .sortBy { case (u: U, _) => u }
+        .sortBy { case (u, _) => u }
```

Comment:
Those give warnings (I think) by the compiler since scala can't see that the type MUST be U, and U is also erased. Scala type inference does not work great with case matches.